src/main/java/nl/tudelft/lifetiles/graph/view/TileView.java

```diff
@@ -36,180 +35,164 @@
-     */
-    private final double spacingY = 1;
-
-    /**
-     * Create the TileView by intilializing the groups where the to be drawn
+     * Create the TileView by initializing the groups where the to be drawn
      * vertices and edges are stored.
      */
     public TileView() {
         root = new Group();
         nodes = new Group();
         edges = new Group();
+        lanes = new LinkedList<Long>();
     }
 
     /**
-     * Add the controller to this view.
+     * Change Vertex colour.
      *
-     * @param tvc
-     *            - the controller
+     * @param v
+     *            vertex to be changed.
+     * @param color
+     *            the new colour
      */
-    public final void addController(final TileController tvc) {
-        controller = tvc;
+    public final void changeVertexColor(final VertexView v, final Color color) {
+        v.setColor(color);
     }
 
     /**
-     * Change Vertex colour.
+     * Draw the given graph.
      *
-     * @param v
-     *            - vertex to be changed.
-     * @param color
-     *            - the new colour
+     * @param gr
+     *            Graph to be drawn
+     * @return the elements that must be displayed on the screen
      */
-    public final void changeVertexColour(final Vertex v, final Color color) {
-        v.setColour(color);
+    public final Group drawGraph(final Graph<SequenceSegment> gr) {
+        PriorityQueue<SequenceSegment> it = sortStartVar(gr);
+        while (!it.isEmpty()) {
+            SequenceSegment segment = it.poll();
+            drawVertexLane(segment);
+        }
+        root.getChildren().addAll(edges, nodes);
+        return root;
     }
 
     /**
-     * Functions that will find if it can draw a segment
-     * at a certain position.
+     * Draws a given segment to an available position in the graph.
      *
      * @param segment
-     *            - segment to be drawn
-     * @param lanes
-     *            - already drawn segments
+     *            segment to be drawn
      */
-    private void checkAvailable(final SequenceSegment segment,
-            final List<Long> lanes) {
-        for (int i = 0; i <= lanes.size(); i++) {
-            if (i >= lanes.size() || lanes.get(i) <= segment.getStart()
-                    && segmentFree(i, segment, lanes)) {
-                segmentInsert(i, segment, lanes);
-                drawVertex(segment.getContent(), segment.getStart(), i, segment
-                        .getSources().size(), Color.GRAY);
+    private void drawVertexLane(final SequenceSegment segment) {
+        for (int index = 0; index <= lanes.size(); index++) {
+            if (index >= lanes.size() || lanes.get(index) <= segment.getStart()
+                    && segmentFree(index, segment)) {
+                segmentInsert(index, segment);
+                String text = segment.getContent().toString();
+                drawVertex(text, segment.getStart(), index, segment
```

Comment:
Bad readability; you should create intermediate variables for the big arguments.